### PR TITLE
Add cardano-testnet queries test: gov-state, stake-distribution

### PR DESCRIPTION
**Pull request by carbolymer**
# Description

Add cardano-testnet queries test: gov-state, stake-distribution

Part of fixing https://github.com/IntersectMBO/cardano-node/issues/5672

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs [hunk 1]
```diff
@@ -78,56 +80,101 @@ hprop_cli_queries = H.integrationWorkspace "cli-queries" $ \tempAbsBasePath' ->
...
-  dreps :: Aeson.Value <- H.noteShowM $ H.execCliStdoutToJson execConfig [ "conway", "query", "drep-state", "--all-dreps"]
-  assertArrayOfSize dreps 3
-  -- query drep-state to a file
-  let drepStateOutFile = work </> "drep-state-out.json"
-  H.noteM_ $ H.execCli' execConfig [ "conway", "query", "drep-state", "--all-dreps"
-                                   , "--out-file", drepStateOutFile]
-  _ :: Aeson.Value <- H.leftFailM . H.readJsonFile $ drepStateOutFile
+
+  -- protocol-parameters
+  do
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs:87]: This `do` is mainly for style purposes right?

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs [hunk 2]
```diff
@@ -78,56 +80,101 @@ hprop_cli_queries = H.integrationWorkspace "cli-queries" $ \tempAbsBasePath' ->
...
+
+  -- tip
+  do
+    -- to stdout
+    _ :: QueryTipLocalStateOutput <- H.noteShowM $ H.execCliStdoutToJson execConfig [ eraName, "query", "tip" ]
+    -- to a file
+    let tipOutFile = work </> "tip-out.json"
+    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "tip"
+                                     , "--out-file", tipOutFile]
+    _ :: QueryTipLocalStateOutput <- H.readJsonFileOk tipOutFile
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs:109]: Should we use a bang pattern here to be sure we are actually decoding the file?

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs [hunk 3]
```diff
@@ -78,56 +80,101 @@ hprop_cli_queries = H.integrationWorkspace "cli-queries" $ \tempAbsBasePath' ->
...
+    -- to a file
+    let stakePoolsOutFile = work </> "stake-pools-out.json"
+    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "stake-pools" , "--out-file", stakePoolsOutFile]
+
+  -- stake-distribution
+  do
+    -- to stdout
+    stakeDistrOut <- H.execCli' execConfig [ eraName, "query", "stake-distribution" ]
+    -- stake addresses with stake
+    let stakeAddresses = map (both T.strip . T.breakOn " " . T.strip . fromString) . drop 2 . lines $ stakeDistrOut
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs:131]: Does this not return JSON?

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs [hunk 4]
```diff
@@ -78,56 +80,101 @@ hprop_cli_queries = H.integrationWorkspace "cli-queries" $ \tempAbsBasePath' ->
...
+    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "drep-state", "--all-dreps"
+                                     , "--out-file", drepStateOutFile]
+    _ :: Aeson.Value <- H.readJsonFileOk drepStateOutFile
+    pure ()
+
 
   H.success
 
+both :: (a -> b) -> (a, a) -> (b, b)
+both f ~(x, y) = (f x, f y)
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs:176]: https://hackage.haskell.org/package/base-4.19.1.0/docs/Data-Bifunctor.html#v:bimap

**Review by Jimbo4350 — COMMENTED** [review]
I'll approve as the last commit looks good but  the first commit needs a couple of changes I described in the other PR.

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs:87]: Yes, to just separate each query

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs (hunk 3 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs:131]: when writing to a file - yes. When writing to stdout, nope, plain text.

Comment (smelc) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs:131]: That's not good, I'll fix that.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs [hunk 5]
```diff
@@ -72,61 +73,108 @@ hprop_cli_queries = H.integrationWorkspace "cli-queries" $ \tempAbsBasePath' ->
...
+    H.assertWith stakeAddresses $ \sa ->
+      -- Because, by default, 3 stake pools are created
+      length sa == 3
+    -- Light test of the query's answer, the ids should exist:
+    forM_ stakeAddresses $ \(stakePoolId, _) -> do
+      H.execCli' execConfig [ eraName, "query", "pool-state"
+                            , "--stake-pool-id", T.unpack stakePoolId ]
+    -- to a file
+    let stakePoolsOutFile = work </> "stake-pools-out.json"
+    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "stake-pools"
```
Comment (smelc) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs:141]:
```suggestion
    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "stake-distribution"
```

@carbolymer> probably a copy/paste mistake here

**Review by smelc — CHANGES_REQUESTED** [review]

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs (hunk 2 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs:109]: That should be done by `readJsonFileOk` which lifts the `Either DecodeError QueryTipLocalStateOutput` to `MonadTest m => m QueryTipLocalStateOutput`, so it should error out at this place.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs (hunk 5 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs]: Good catch! Thanks, fixed!

**Comment by carbolymer** [comment]
Macos & windows failures are unrelated to this PR - there's an issue with the base image.

**Pull request merged**